system_metrics_collector/test/topic_statistics_collector/test_subscriber_topic_statistics.cpp

```diff
@@ -0,0 +1,346 @@
+{
+public:
+  void SetUp() override
+  {
+    rclcpp::init(0, nullptr);
+
+    rclcpp::NodeOptions options;
+    options.append_parameter_override(kPublishPeriodParam, kDontPublishDuringTest.count());
+
+    test_periodic_publisher_ = std::make_shared<TestSubscriberTopicStatistics>(
+      kTestNodeName, options);
+
+    const moving_average_statistics::StatisticData data =
+      test_periodic_publisher_->GetStatisticsResults();
+    ASSERT_TRUE(std::isnan(data.average));
+    ASSERT_TRUE(std::isnan(data.min));
+    ASSERT_TRUE(std::isnan(data.max));
+    ASSERT_TRUE(std::isnan(data.standard_deviation));
+    ASSERT_EQ(0, data.sample_count);
+  }
+
+  void TearDown() override
+  {
+    test_periodic_publisher_->shutdown();
+    EXPECT_EQ(State::PRIMARY_STATE_FINALIZED, test_periodic_publisher_->get_current_state().id());
+    EXPECT_FALSE(test_periodic_publisher_->IsStarted());
+    EXPECT_FALSE(test_periodic_publisher_->IsPublisherActivated());
+
+    test_periodic_publisher_.reset();
+    rclcpp::shutdown();
+  }
+
+protected:
+  // this test is not designed to have the statistics published and reset at any point of the test,
+  // so this is defining the publish period to be something amply larger than the test duration
+  // itself
+  static constexpr std::chrono::milliseconds kDontPublishDuringTest = 2 * kTestDuration;
+  std::shared_ptr<TestSubscriberTopicStatistics> test_periodic_publisher_;
+};
+
+/**
+ * Fixture to bringup and teardown rclcpp
+ */
+class RclcppFixture : public ::testing::Test
+{
+public:
+  void SetUp() override
+  {
+    rclcpp::init(0, nullptr);
+  }
+
+  void TearDown() override
+  {
+    rclcpp::shutdown();
+  }
+};
+
+std::shared_ptr<sensor_msgs::msg::Imu> GetImuMessageWithHeader(const int64_t timestamp)
+{
+  auto message = sensor_msgs::msg::Imu{};
+  message.header = std_msgs::msg::Header{};
+  message.header.stamp = rclcpp::Time{timestamp};
+  return std::make_shared<sensor_msgs::msg::Imu>(message);
+}
+
+constexpr std::chrono::milliseconds SubscriberTopicStatisticsTestFixture::kDontPublishDuringTest;
+
+TEST_F(SubscriberTopicStatisticsTestFixture, Sanity) {
+  ASSERT_NE(test_periodic_publisher_, nullptr);
+  ASSERT_EQ(
+    State::PRIMARY_STATE_UNCONFIGURED,
+    test_periodic_publisher_->get_current_state().id());
+
+  test_periodic_publisher_->configure();
+  ASSERT_EQ(State::PRIMARY_STATE_INACTIVE, test_periodic_publisher_->get_current_state().id());
+  ASSERT_FALSE(test_periodic_publisher_->IsPublisherActivated());
+
+  test_periodic_publisher_->activate();
+  ASSERT_TRUE(test_periodic_publisher_->IsStarted());
+  ASSERT_EQ(State::PRIMARY_STATE_ACTIVE, test_periodic_publisher_->get_current_state().id());
+  ASSERT_TRUE(test_periodic_publisher_->IsPublisherActivated());
+
+
+  ASSERT_EQ(
+    "name=test_periodic_node, publishing_topic=/system_metrics,"
+    " publish_period=500ms, started=true,"
+    " avg=nan, min=nan, max=nan, std_dev=nan, count=0",
+    test_periodic_publisher_->GetStatusString());
```

Comment:
Consider a `stringstream` with expected values as constants to make this assertion more reliable.